From f2b10db5fbe0156374c26b2999ff27e0f639c225 Mon Sep 17 00:00:00 2001
From: Alon Bar-Lev <alon.barlev@gmail.com>
Date: Wed, 10 Oct 2012 16:54:08 +0000
Subject: [PATCH] build: allow disable iconv for static

Signed-off-by: Alon Bar-Lev <alon.barlev@gmail.com>
---
 password.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/password.c b/password.c
index ac58082..70310b2 100644
--- a/password.c
+++ b/password.c
@@ -223,6 +223,15 @@ int passwd_to_utf16(char *in_passwd,
                     int max_length,
                     char *out_passwd)
 {
+#ifdef DISABLE_ICONV
+    /* support only latin */
+    int i;
+    for (i=0;i<length+1;i++) {
+    	out_passwd[i*2] = in_passwd[i];
+	out_passwd[i*2+1] = 0;
+    }
+    return length*2;
+#else
     char *ic_outbuf,
          *ic_inbuf;
     iconv_t condesc;
@@ -260,5 +269,6 @@ int passwd_to_utf16(char *in_passwd,
     }
     iconv_close(condesc);
     return (max_length - ic_outbytesleft);
+#endif
 }
 
-- 
1.7.8.6

